setup-branch-protection.sh

```
@@ -0,0 +1,97 @@
+
+# Get repository info
+REPO_OWNER=$(gh repo view --json owner --jq '.owner.login')
+REPO_NAME=$(gh repo view --json name --jq '.name')
+DEFAULT_BRANCH=$(gh api repos/$REPO_OWNER/$REPO_NAME --jq '.default_branch')
+
+echo "📋 Repository: $REPO_OWNER/$REPO_NAME"
+echo "🌿 Default branch: $DEFAULT_BRANCH"
+
+# Check if user has admin permissions
+echo "🔐 Checking permissions..."
+if ! gh api repos/$REPO_OWNER/$REPO_NAME/collaborators/$(gh api user --jq '.login')/permission --jq '.permission' | grep -q "admin"; then
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)